### PR TITLE
Add range iterator to LocalVector

### PR DESCRIPTION
**Pull request by KoBeWi**
In this edition of *enormous PR that touches every area of the engine* I'm adding a range iterator to LocalVector type. I went ahead and also replaced all old loops where I found it applicable.

### CODE REVIEW

File: core/math/geometry_3d.cpp [hunk 1]
```diff
@@ -141,21 +141,19 @@ real_t Geometry3D::get_closest_distance_between_segments(const Vector3 &p_p0, co
 void Geometry3D::MeshData::optimize_vertices() {
 	HashMap<int, int> vtx_remap;
 
-	for (uint32_t i = 0; i < faces.size(); i++) {
-		for (uint32_t j = 0; j < faces[i].indices.size(); j++) {
-			int idx = faces[i].indices[j];
-			if (!vtx_remap.has(idx)) {
+	for (MeshData::Face &face : faces) {
+		for (int &index : face.indices) {
```
Comment (akien-mga) [inline core/math/geometry_3d.cpp:145]: I don't know if it makes much sense to take the int as a reference.

**Review by akien-mga — APPROVED** [review]
That looks really good! I haven't reviewed in depth but we could merge after beta11 is released to have a week to catch potential regressions.

**Comment by lmurray** [comment]
A quick drive-by review: For a vector-like class it is possible to just return a raw pointer for `begin()` and `end()` without the need of an actual iterator class. This is how the STL usually implements the iterator for std::vector. As it stands the current iterator classes are just a passthrough and might get in the way of the optimizer.

**Review by reduz — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**